journal/src/test/java/io/zeebe/journal/file/JournalReaderTest.java

```diff
@@ -141,6 +141,83 @@ public void shouldReadAppendedDataAfterSeek() {
     assertThat(reader.hasNext()).isTrue();
   }
 
+  @Test
+  public void shouldSeekToAsqn() {
+    // given
+    long asqn = 10;
+    JournalRecord lastRecordWritten = null;
+    for (int i = 1; i <= ENTRIES_PER_SEGMENT * 2; i++) {
+      final JournalRecord record = journal.append(asqn++, data);
+      assertThat(record.index()).isEqualTo(i);
+      lastRecordWritten = record;
+    }
+    assertThat(reader.hasNext()).isTrue();
+
+    // when
+    reader.seekToAsqn(lastRecordWritten.asqn() - 2);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    assertThat(reader.next().asqn()).isEqualTo(lastRecordWritten.asqn() - 2);
+  }
+
+  @Test
+  public void shouldSeekToHighestAsqnLowerThanProvidedAsqn() {
+    // given
+    final var expectedRecord = journal.append(1, data);
+    journal.append(5, data);
+
+    // when
+    reader.seekToAsqn(4);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    final var record = reader.next();
+    assertThat(record.index()).isEqualTo(expectedRecord.index());
+    assertThat(record.asqn()).isEqualTo(expectedRecord.asqn());
+    assertThat(record.data()).isEqualTo(expectedRecord.data());
+    assertThat(record.checksum()).isEqualTo(expectedRecord.checksum());
+  }
+
+  @Test
+  public void shouldSeekToHighestLowerAsqnEvenIfRecordHasNone() {
```

Comment:
No, this checks that it returns the record the highest record event if it isn't mapped from asqn to index.

Reply:
But that is not the expected behavior. It should seek to the record with asqn 1. That means, I think the current implementation is not correct. You can mark this record with @Ignore and create an issue to fix it.

Reply:
> But that is not the expected behavior.

The interface's documentation doesn't say anything about that, maybe we should clarify that as well.

@npepinpe turns out what we discussed earlier is already the expected behavior and the implementation is wrong. Fixing this would make your PR simpler

Reply:
Interface documentation says `Seek to a record with the highest asqn less than or equal to the given asqn.`  If it is not clear, feel free to update the doc.

Reply:
Nicolas is already doing it apparently